sdk/appconfiguration/app-configuration/test/internal/http.spec.ts

```diff
@@ -104,14 +104,63 @@ describe("http request related tests", () => {
     });
   });
 
+  describe("custom client ID", () => {
+    let client: AppConfigurationClient;
+    let scope: nock.Scope;
+
+    beforeEach(function() {
+      client = createAppConfigurationClientForTests() || this.skip();
+      scope = nock(/.*/);
+    });
+
+    afterEach(() => {
+      assert.ok(scope.isDone());
+      nock.cleanAll();
+    });    
+
+    it("custom client request ID", async () => {
+      scope
+        .matchHeader('x-ms-client-request-id', /this is my custom client request id/)
+        .get(/.*/)
+        .reply(200);
+      
+      const iterator = await client.listConfigurationSettings({
+        requestOptions: {
+          customHeaders: {
+            "x-ms-client-request-id": "this is my custom client request id"
+          }
+        }
+      });
+
+      await iterator.next();
+    });
+
+    it("default client request ID", async () => {
+      scope
+        .matchHeader('x-ms-client-request-id', /[A-Za-z0-9\-]+/)
+        .get(/.*/)
```

Comment:
Sorry I'm not too familiar with nock. What does this method call do?

Reply:
Basically:

For a get request to any URL that matches this regex (/.*/) make sure it passes an 'x-ms-client-request-id' header that matches the following regex blah.

I have two of these matchers - the default one (this particular test case) is an actual UUID so I just validate that it's not empty and was filled in.